src/Core/Cart/CartRow.php

```diff
@@ -284,6 +284,12 @@ public function processCalculation(Cart $cart)
                 $tools->round($this->initialUnitPrice->getTaxIncluded(), $this->precision) * $quantity,
                 $tools->round($this->initialUnitPrice->getTaxExcluded(), $this->precision) * $quantity
             );
+        } elseif ($this->roundType == self::ROUND_MODE_LINE) {
+            $tools = new Tools();
```

Comment:
`$tools` is already used in line 282. I think you should change this to declare it only once, before the `if()`.

Reply:
Done